### PR TITLE
feat: 🎸 update headlessui v2

### PR DESCRIPTION
**Pull request by kodai3**
# Changes
https://ubie.atlassian.net/browse/DSGNPB-273?atlOrigin=eyJpIjoiYTVhMDUwODUwMDM4NDhmYjk4ZThhM2I1MDBlYThlYTYiLCJwIjoiamlyYS1zbGFjay1pbnQifQ

# Check

- [x] Browser verification (minimum) Android Chrome/iOS Safari(375px-)
- [x] CSS not affected by inheritance
- [x] Layout does not break even if there is an overflow
- [x] Layout does not break when wraps
- [x] Added new Component
  - [ ] Added data-* prop and id prop
- [x] Updated [Ubie Vitals](https://github.com/ubie-oss/ubie-vitals-website) or Added an update [issue](https://github.com/ubie-oss/ubie-vitals-website/issues)(if needed)

### CODE REVIEW

File: src/components/ActionHalfModal/ActionHalfModal.tsx [hunk 1]
```diff
@@ -164,9 +164,9 @@ export const ActionHalfModal: FC<Props> = ({
           leaveFrom={styles.overlayLeaveFrom}
           leaveTo={styles.overlayLeaveTo}
         >
-          <Dialog.Overlay className={clsx(styles.overlay, styles[opacityClassName])} />
```
Comment (kodai3) [inline src/components/ActionHalfModal/ActionHalfModal.tsx:167]: > If you still run into this, you should be able to replace the `<DialogOverlay>` and `<DialogBackdrop>` components with a simple `<div>` instead.



https://github.com/tailwindlabs/headlessui/pull/3171

File: package.json [hunk 2]
```diff
@@ -125,12 +125,12 @@
...
   "peerDependencies": {
-    "@headlessui/react": ">1.7.0 <2.0.0",
+    "@headlessui/react": "^2.0.0",
     "@ubie/design-tokens": ">=0.2.0",
     "@ubie/ubie-icons": ">=0.5.0 <0.6.2 || >=0.8.0",
     "clsx": ">1.2.0",
-    "react": "^17 || ^18",
-    "react-dom": "^17 || ^18"
+    "react": "^18",
+    "react-dom": "^18"
```
Comment (kodai3) [inline package.json:133]: https://github.com/tailwindlabs/headlessui/pull/3131

**Comment by 8845musign** [comment]
Thanks for the implementation🙏 
Because it overlaps with https://github.com/ubie-oss/ubie-ui/pull/134, I will FB it to #134 after looking at the content.

**Comment by 8845musign** [comment]
https://github.com/ubie-oss/ubie-ui/pull/134 is meged.

**Pull request closed without merging**